climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1350 @@
+        from the intensity values of the historical fire
+
+        Parameters
+        ----------
+            self : climada.hazard.WildFire instance
+            centr_burned : np.array
+                array indicating which centroids burned
+
+        Returns
+        -------
+            proba_intensity : lil_matrix
+                hazard intensity matrix of generated probabilistic fire
+
+        """
+        # The brightness values are chosen randomly at every burned centroids
+        # from the brightness values of the historical fire
+        ev_proba_uni = centr_burned.nonzero()[0] * self.centroids.shape[1] + \
+            centr_burned.nonzero()[1]
+        proba_intensity = sparse.lil_matrix(np.zeros((1, self.centroids.size)))
+        for ev_prob in ev_proba_uni:
+            proba_intensity[0, ev_prob] = np.random.choice( \
+                self.intensity.data)
+
+        return proba_intensity
+
+    def _set_fire_propa_matrix(self, n_blurr=BLURR_STEPS):
+
+        """ sets fire propagation matrix which is used to propagate
+        probabilistic fires. The matrix is set so that burn probability on
+        centroids which burned historically is set to 1. A blurr with
+        exponential decay of burn probabilities is set around these
+        centroids.
+        Alternatively, the fire propagation probability matrix can be any
+        matrix that coresponds to the shape of the centroids and thus not have
+        to be set this way.
+
+        Parameters
+        ----------
+            self : climada.hazard.WildFire instance
+            n_blurr : int
+                blurr width around historical fires
+
+        Returns
+        -------
+            self.centroids.fire_propa_matrix : np.array
+
+        """
+        # historically burned centroids
+        hist_burned = np.zeros(self.centroids.lat.shape, dtype=bool)
+        hist_burned = self.intensity.sum(0) > 0.
+        self.centroids.hist_burned = hist_burned
+
+        fire_propa_matrix = hist_burned.reshape(self.centroids.shape).astype(float)
+        blurr_lvl = np.ones(n_blurr)
+        # exponential decay of fire propagation on cell level
+        for i in range(n_blurr):
+            blurr_lvl[i] = 2.**(-i)
+        # Neighbourhood
+        hood = ((-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1))
+
+        # loop over matrix to create blurr around historical fires
+        for blurr in range(n_blurr-1):
+            for i in range(fire_propa_matrix.shape[0]):
+                for j in range(fire_propa_matrix.shape[1]):
+                    if fire_propa_matrix[i, j] == blurr_lvl[blurr]:
+                        for _, (delta_x, delta_y) in enumerate(hood):
+                            try:
+                                if fire_propa_matrix[i+delta_x, j+delta_y] == 0:
+                                    fire_propa_matrix[i+delta_x, j+delta_y] = blurr_lvl[blurr+1]
```

Comment:
In case efficiency is an issue in the module, i have the feeling that this part here could at least partly be vectorized.
TBD.

Reply:
I won't change this now, as I'll be working with Sam the Master student on this. But I'll keep it in mind!

Reply:
I won't change this now, as I'll be working with Sam the Master student on this. But I'll keep it in mind!